### DISCUSSION TITLE
2022/05/18/ssh-passphrases-kde/

### DISCUSSION THREAD

**Discussion by giscus** [jdeluyck/kcoreorg-website#147] [Blog Comments]
# 2022/05/18/ssh-passphrases-kde/

I finally (shame on me) cycled all my SSH keys to use passphrases. One thing I massively dislike is having to enter my passphrase when using them. ssh-agent to the rescue!

https://kcore.org/2022/05/18/ssh-passphrases-kde/

**Comment by v1rtl** [comment]
Thanks for the article.

I've noticed a few issues:

For `ssh-agent-startup.sh` it should be `#/!bin/bash` instead of `#!bin/bash`

Running the automatic key import is not gonna work because there's no trailing `_` at the end of keys, it should be this instead:

```sh
#!/bin/bash

# Wait for kwallet
kwallet-query -l kdewallet > /dev/null

for KEY in $(ls $HOME/.ssh/id_rsa* | grep -v \.pub); do
  ssh-add -q ${KEY} </dev/null
done

for KEY in $(ls $HOME/.ssh/id_ed25519* | grep -v \.pub); do
  ssh-add -q ${KEY} </dev/null
done
```

**Reply by jdeluyck** [reply]
Hi,

Thanks for this. I've added the missing `/` in the shell.
For the other thing - the missing _ - that is really a user thing. My keys do have id_<key type>_<description>, so for me this works.

**Comment by Roy-Orbison** [comment]
Thanks for this write-up. I had to change the process search to be more specific, otherwise it matched previously killed instances that sometimes hang around as zombies, and a `/usr/libexec/gcr-ssh-agent` process:
```bash
pgrep --euid "$USER" --runstates DRS --exact ssh-agent
```

I also changed the key adding to something like this to avoid both quoting issues, and parsing `ls`:
```bash
for KEY in "$HOME"/.ssh/id_{rsa,ed25519}_*; do
	if [[ "$KEY" != *.pub ]]; then
		ssh-add -q "$KEY" < /dev/null
	fi
done
```
